community/values/src/main/java/org/neo4j/values/storable/UTF8StringValue.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values.storable;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/*
+ * Just as a normal StringValue but is backed by a byte array and does string
+ * serialization lazily when necessary.
+ *
+ */
+public final class UTF8StringValue extends StringValue
+{
+    //0111 1111, used for removing HIGH BIT from byte
+    private static final int HIGH_BIT_MASK = 127;
+    //0100 000, used for detecting non-continuation bytes, e.g. 10xx xxxx
+    private static final int NON_CONTINUATION_BIT_MASK = 64;
```

Comment:
Same thing here, I think this would look better using a javadoc comment and a binary literal:

    /** Used for detecting non-continuation bytes. For example {@code 0b10xx_xxxx}. */
    private static final int NON_CONTINUATION_BIT_MASK = 0b0100_0000;